python/cudf/dataframe/multiindex.py

```diff
@@ -200,9 +246,16 @@ def _get_column_major(self, df, row_tuple):
         return result
 
     def __len__(self):
+        if self._codes is None:
+            if self._df is not None:
+                self._compute_levels_and_codes()
+            else:
+                return 0
         return len(self.codes[self.codes.columns[0]])
 
     def __eq__(self, other):
+        if self._codes is None:
+            self._compute_levels_and_codes()
         if not hasattr(other, 'levels'):
             return False
         equal_levels = self.levels == other.levels
```

Comment:
Instead of checking the levels and codes here, we can just do a series of binary ops against the actual columns

Reply:
Interesting idea